### PR TITLE
Add react-storybook module definition

### PR DESCRIPTION
**Pull request by ajhyndman**
Hey.  I put this together for my own project, and thought it might be useful for others.

I tried to follow the definition and test style from other packages.  Am I missing anything?

### CODE REVIEW

**Comment by gabelevi** [comment]
The definition looks good, but I wonder about the file path. This seems to be the first scoped npm package in flow-typed, as far as I can tell. Since the file path has semantic meaning attached, 

```
definitions/npm/react-storybook_v1.x.x/flow_>=v0.30.x/react-storybook_v1.x.x.js
```

is probably not the right name. That said, I'm not sure if there's a way to represent a scoped package, at the moment. We'd probably need to escape the backslash somehow, or support nested directories. Maybe something like

```
definitions/npm/@kadira/storybook_v1.x.x/flow_>=v0.30.x/storybook_v1.x.x.js
```

I'll ping @jeffmo when he's back next week and see what he says. Sorry for the delay, but thanks for this PR and bringing scopes to our attention!

**Comment by ajhyndman** [comment]
@gabelevi sounds good.  Thanks for taking a look!

File: definitions/npm/react-storybook_v1.x.x/flow_>=v0.30.x/react-storybook_v1.x.x.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+declare module '@kadira/storybook' {
+  declare interface Story {
+    add: (storyName: string, callback: Function) => Story;
+  }
+
+  declare function storiesOf(name: string, module: any): Story;
+  declare function action(name: string): Function;
+}
```
Comment (thejameskyle) [inline definitions/npm/react-storybook_v1.x.x/flow_>=v0.30.x/react-storybook_v1.x.x.js]: Can you add the entire API interface? Looks like there is all of this: https://github.com/kadirahq/react-storybook/blob/master/src/client/index.js#L3-L11

Also, a couple things that would make this better:

- Specify the callbacks more than just `Function`, i.e. `(..., callback: (arg: boolean) => string) => Story`
- Specify what the `any`'s are. Either the specific type or use `mixed`.

**Comment by jeffmo** [comment]
> Maybe something like
> 
```
definitions/npm/@kadira/storybook_v1.x.x/flow_>=v0.30.x/storybook_v1.x.x.js
```

Yea, that makes sense to me and I think it'd work.

Will require an update to the code to understand the path meaning, but shouldn't be too difficult. We'll probably want to coordinate this change with https://github.com/flowtype/flow-typed/issues/225 (cc @samwgoldman). Sam, do you have a sense for when you'll have something ready there?

If soon in the next week, I'll just hold off (I don't think this change will be super difficult, but it might be merge-conflicty). Otherwise I can probably add support for this today.

**Comment by samwgoldman** [comment]
I'm working through something right now that might take a few days. Go ahead and I'll deal with rebase

**Comment by jeffmo** [comment]
Over to @avikchaudhuri who will pick this up while I work on docs

**Comment by jeffmo** [comment]
Ok, [now that we have support for scoped npm packages](https://github.com/flowtype/flow-typed/pull/303) (woo! Thanks @avikchaudhuri) would you mind updating the location of the libdef in this PR?

**Comment by ajhyndman** [comment]
I'm going to be honest, I have other pressing priorities for the next month or so.  I won't have a lot of time to improve the API declaration coverage.

I'm happy to update the path when I get home tonight, though.  How would you like to proceed with the rest?

**Comment by ajhyndman** [comment]
It seems like the Travis build died.  I'm going to attempt to close and re-open the PR to trigger a re-run.

**Comment by ajhyndman** [comment]
No luck.  I might leave it for someone who knows what's going on to sort out.  :(

**Comment by jeffmo** [comment]
Sorry about this! I've restarted the build (and also hopefully fixed the issue for future PRs [over here](https://github.com/flowtype/flow-typed/commit/5ba653a28a0bec1f30fc909c77d4771d93bbb703))

**Pull request merged**